File: app/src/main/java/net/janczar/powertape/example/view/base/BasePresenter.java
package net.janczar.powertape.example.view.base;


import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public class BasePresenter<T extends BaseView> {

    private final CompositeDisposable disposables = new CompositeDisposable();

    protected T view;

    public void bind(T view) {
        this.view = view;
    }

    public void unbind() {
        this.view = null;
        disposables.dispose();
    }

    protected void manage(Disposable disposable) {
        disposables.add(disposable);
    }
}

File: app/src/main/java/net/janczar/powertape/example/view/base/BasePresenter.kt
package net.janczar.powertape.example.view.base

import android.support.annotation.CallSuper
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable


open class BasePresenter<V : BaseView> {

    private var compositeDisposable = CompositeDisposable()

    protected var view: V? = null

    fun attachView(view: V) {
        this.view = view
        onAttach()
    }

    @CallSuper
    open fun detachView() {
        compositeDisposable.clear()
        onDetach()
    }

    protected fun manage(disposable: Disposable) {
        compositeDisposable.add(disposable)
    }

    open protected fun onAttach() = Unit

    open protected fun onDetach() = Unit

}